### ISSUE TITLE
The NB does something wrong with enumeration parameters

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11908]
Please check the `PowerGrids.Electrical.Test.SynchronousMachine4Windings` test case in PowerGrids 1.0.3.

The test on the master branch produced the [correct results](https://libraries.openmodelica.org/branches/master/PowerGrids/PowerGrids.html), with 148 verified signals.

The test on the newInst-newBackend branch [fails the verification](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/PowerGrids.html) on [5/148 signals](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/files/PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4Windings.diff.html). Two of them, [machine.localInit](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/files/PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4Windings.diff.machine.localInit.html) and [machine.initOpt](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/files/PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4Windings.diff.machine.initOpt.html) are enumeration parameters. In both cases, their values correspond to the first element of the enumeration, so they should both be 1, but the simulation result with the NB has value 0 instead. This also has consequences on some numerical values.

I guess something goes wrong somewhere with their processing, as enumerations should never get a value of zero. This simple MWE
```modelica
model TestEnum
  type T = enumeration(none, some, all);
  
  final parameter T t1 = T.none;
  final parameter T t2 = t1;
  final parameter T t3 = T.all;
end TestEnum;
```
works fine with --newBackend, so I guess it's something a bit more subtle.

**Comment by phannebohm** [comment]
Might be fixed by #12436, please check. We forgot to copy all variable attributes from the NB to SimCode.

**Comment by phannebohm** [comment]
The model mentioned here currently fails during Tearing.

The original issue should be fixed in #12436, though. Start values of fixed enumeration variables were not properly carried to simcode. That led to the variables not getting initialized properly and thus having a value of zero.

There were a number of models not verifying with enumeration variables. After the last [test run](https://libraries.openmodelica.org/branches/history/newInst-newBackend/2024-05-17%2016:01:18..2024-05-20%2011:00:01.html) I could not find any enumerations that don't verify.

**Comment by casella** [comment]
> The model mentioned here currently fails during Tearing.

Yes, that is described in #12424: mix-up between parameters with fixed = false and discrete variables during initialization.

> The original issue should be fixed in #12436, though. Start values of fixed enumeration variables were not properly carried to simcode. That led to the variables not getting initialized properly and thus having a value of zero.

Good!

> There were a number of models not verifying with enumeration variables. After the last [test run](https://libraries.openmodelica.org/branches/history/newInst-newBackend/2024-05-17%2016:01:18..2024-05-20%2011:00:01.html) I could not find any enumerations that don't verify.

Excellent. One down 😄
